packages/client/hmi-client/src/components/model/petrinet/tera-distribution-input.vue

```diff
@@ -0,0 +1,197 @@
+		<template v-for="(option, index) in options" :key="index">
+			<tera-input-number
+				class="parameter-input"
+				error-empty
+				:label="DistributionInputLabels[parameter.type][index]"
+				:model-value="getParameterDistribution(modelConfiguration, parameterId, true)?.parameters[option]"
+				@update:model-value="onParameterChange($event, DistributionInputOptions[parameter.type][index])"
+			/>
+		</template>
+	</span>
+</template>
+<script setup lang="ts">
+import { computed, ref, onMounted, nextTick } from 'vue';
+import { getParameterDistribution, isNumberInputEmpty } from '@/services/model-configurations';
+import { Model, ModelConfiguration, ModelDistribution } from '@/types/Types';
+import Dropdown from 'primevue/dropdown';
+import TeraInputNumber from '@/components/widgets/tera-input-number.vue';
+import {
+	DistributionType,
+	externaDoclLinks,
+	distributionTypeOptions,
+	DistributionInputOptions,
+	DistributionInputLabels,
+	distributionDescription,
+	distributionParamDescription
+} from '@/services/distribution';
+import Card from 'primevue/card';
+
+const props = defineProps<{
+	model: Model;
+	modelConfiguration: ModelConfiguration;
+	parameterId: string;
+	parameter: ModelDistribution;
+}>();
+
+const emit = defineEmits(['update-parameter', 'update-source']);
+
+const options = computed(() => DistributionInputOptions[props.parameter.type]);
+
+const dropdownRef = ref();
+const cardRef = ref();
+const isParameterEmpty = ref<boolean>(false);
+const distributionOptions = ref<{ name: string; value: string }[]>(distributionTypeOptions());
+const hoveredOption = ref<{ name: string; value: string } | null>();
+const popupPosition = ref({ top: 0, left: 0 });
+
+function hidePopup() {
+	hoveredOption.value = null;
+}
+
+const cardElement = computed(() => cardRef.value.$el.getBoundingClientRect());
+
+async function showPopup(item: HTMLElement | null, index: number) {
+	hidePopup();
+	if (!item) return;
+
+	hoveredOption.value = distributionOptions.value[index];
+	await nextTick();
+	const rect = item.getBoundingClientRect();
+	const viewport = window.innerHeight;
+	const cardBottom = cardElement.value.height + rect.bottom;
+
+	let top = rect.y / rect.top + window.scrollY - 35;
+	if (cardBottom > viewport) {
+		top -= Math.abs(cardBottom - viewport) + 35;
+	}
+
+	popupPosition.value = {
+		top,
+		left: rect.x / rect.left + rect.width
+	};
+}
+
+function onChange(event) {
+	emit('update-parameter', {
+		id: props.parameterId,
+		distribution: formatPayloadFromTypeChange(event.value)
+	});
+	hidePopup();
+}
+
+function isParameterInputEmpty(parameter) {
+	if (!DistributionInputOptions?.[parameter.type]?.length) {
+		return true;
+	}
+
+	if (DistributionInputOptions[parameter.type].length === 1) {
+		return isNumberInputEmpty(DistributionInputOptions[parameter.type][0]);
+	}
+	return (
+		isNumberInputEmpty(DistributionInputOptions[parameter.type][0]) ||
+		isNumberInputEmpty(DistributionInputOptions[parameter.type][1])
+	);
```

Comment:
The explicit if block check for `if (DistributionInputOptions[parameter.type].length === 1)` seems redundant because it is covered by the other one.

What logic are you trying to achieve here?

Reply:
removed unneeded code